lib/query-interface.js

```diff
@@ -217,7 +230,7 @@ class QueryInterface {
                     valueOptions.after = vals[idx - 1];
                   }
                   valueOptions.supportsSearchPath = false;
-                  promises.push(this.sequelize.query(this.QueryGenerator.pgEnumAdd(tableName, keys[i], value, valueOptions), valueOptions));
+                  promises.push(this.sequelize.query(this.QueryGenerator.pgEnumAdd(tableName, attribute.field || keys[i], value, valueOptions), valueOptions));
```

Comment:
This will fix https://github.com/sequelize/sequelize/issues/7812, field name was not used here